freelex.py

```diff
@@ -27,18 +27,25 @@ def fetch_assets(root):
                     with open(fn, "wb") as f:
                         f.write(r.read())
 
-def rename_assets(root):
 # Modify filenames to match the Android requirements (lowercase a-z and _ only)
+# Since iOS uses the same data source (the .dat file), update iOS to use the same image names.
+def normalize_image_filename(filename):
+    normalized_filename = filename.replace('-', '_').lower()
+    num_of_periods = normalized_filename.count('.')
+    if (num_of_periods > 1):
+        normalized_filename = normalized_filename.replace('.', '_', num_of_periods - 1)
+
+    return normalized_filename
+
+
+def rename_assets(root):
      for entry in root.iter("entry"):
          for asset in entry.find("ASSET"):
              if ("picture" == asset.tag):
-                 oldfn = os.path.join(asset.tag, asset.text)
-                 newfn = oldfn.replace('-', '_').lower()
-                 num_of_periods = newfn.count('.')
-                 if (num_of_periods > 1):
-                     newfn = newfn.replace('.', '_', num_of_periods - 1)
-                 os.rename(oldfn, newfn)
-                 asset.text = newfn.replace('picture/', '', 1)
+                 old_filename = os.path.join(asset.tag, asset.text)
+                 new_filename = normalize_image_filename(old_filename)
+                 os.rename(old_filename, new_filename)
+                 asset.text = new_filename.replace('picture/', '', 1)
```

Comment:
This is much easier to read. Is a check that the file exists a good idea here? or a catch of the exception if it doesn't so it'll continue the loop.

Reply:
:elephant:

Reply:
I've manged some Python together to address your suggestion 🐍